saplings/oauth-login/src/index.js

```diff
@@ -34,12 +34,12 @@ registerConfigSapling('login', () => {
     const accessTokenRegex = /\?access_token=([^&#]*)|&|$/;
     const token = window.$CANOPY.redirectedFrom.match(accessTokenRegex)[1];
     // Extract user information
-    const userIdRegex = /\?user_id=([^&#]*)|&|$/;
+    const userIdRegex = /&user_id=([^&#]*)|&|$/;
```

Comment:
These regexes assume a particular order to the query parameters; they should probably allow for them to be in any order (each could be preceded by `?` or `&`).